Classes/common/HTTP/CDTURLSessionTask.h

```diff
@@ -17,10 +17,20 @@
 #import <Foundation/Foundation.h>
 #import "CDTMacros.h"
 
+@class CDTURLSession;
+
+@protocol CDTURLSessionTaskDelegate
+- (void)handleData:(nullable NSData *)data;
+- (void)handleResponse:(nullable NSURLResponse *)response;
+- (void)handleError:(nullable NSError *)error;
+@end
+
 @interface CDTURLSessionTask : NSObject
+{
+    id <CDTURLSessionTaskDelegate> _delegate;
```

Comment:
I'm confused as to why you are doing this, shouldn't it be left up to the compiler to create the variable via the `delegate` property?